lib/jekyll/renderer.rb

```diff
@@ -67,6 +67,10 @@ def run
       output = convert(output)
       document.content = output
 
+      # Hook to modify converted content before layout is applied
+      document.trigger_hooks(:post_convert, output)
+      output = document.content
```

Comment:
I concur, we shouldn't assume users are destructive and should instead encourage them to return their modified content, especially since we've an open ticket to start freezing all our strings, this will certainly impede that and delay it much further.